octoprint_mrbeam/software_update_information.py

```diff
@@ -98,233 +228,190 @@ def switch_software_channel(plugin, channel):
+                        tier=get_tier_by_id(tier)
+                    )
+
+                # get version number
+                current_version = "-"
+                _logger.debug(
+                    "pip command check %s %s - %s",
+                    module,
+                    moduleconfig,
+                    "pip_command" not in moduleconfig,
+                )
+                if (
+                        "global_pip_command" in module
+                        and "pip_command" not in moduleconfig
+                ):
+                    moduleconfig["pip_command"] = GLOBAL_PIP_COMMAND
+                if "pip_command" in moduleconfig:
+                    #get version number of pip modules
+                    pip_command = moduleconfig["pip_command"]
+                    # if global_pip_command is set module is installed outside of our virtualenv therefor we can't use default pip command.
+                    # /usr/local/lib/python2.7/dist-packages must be writable for pi user otherwise OctoPrint won't accept this as a valid pip command
+                    # pip_command = GLOBAL_PIP_COMMAND
+                    package_name = (
+                        module["package_name"]
+                        if "package_name" in module
+                        else module_id
+                    )
+                    _logger.debug("get version %s %s", package_name, pip_command)
+
+                    current_version_global_pip = get_version_of_pip_module(
+                        package_name, pip_command
+                    )
+                    if current_version_global_pip is not None:
+                        current_version = current_version_global_pip
+
+                else:
+                    # get versionnumber of octoprint plugin
+                    pluginInfo = plugin._plugin_manager.get_plugin_info(module_id)
+                    if pluginInfo is not None:
+                        current_version = pluginInfo.version
+
+                if module_id != 'octoprint':
+                    _logger.debug("%s current version: %s", module_id, current_version)
+                    moduleconfig.update(
+                        {
+                            "displayVersion": current_version,
+                        }
+                    )
+                if "name" in module:
+                    moduleconfig["displayName"] = module["name"]
+                # sw_update_config.update(moduleconfig)
+                moduleconfig = clean_update_config(moduleconfig)
+                sw_update_config[module_id] = moduleconfig
+
+        _logger.debug("sw_update_config {}".format(sw_update_config))
+        return sw_update_config
+
+def clean_update_config(update_config):
+    pop_list = ["alpha", "beta", "stable", "develop", "beamos_date", "name"]
+    for pop_element in pop_list:
+        if pop_element in update_config:
+            update_config.pop(pop_element)
```

Comment:
you could also `del(update_config[elm]) for elm in pop_list if elm in update_config`

Reply:
this is not working

Reply:
sorry I meant to remove the parenthesis for the `del` keyword. I corrected my comment.

`del` is more specific for removing pointers.

Reply:
but you can also make it simpler with
```python
for key in set(update_config).intersection(pop_list):
  del update_config[key]
```